### PR TITLE
IDEMPIERE-5567 Support of UUID as Key (FHCA-4195) - Implement support for Requests on UUID Based Tables

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5567

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [x] I have tested the direct scenario that my code is solving
- [x] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [x] I have made corresponding changes to the documentation as follows:
- - [x] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [x] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/MRequest.java [hunk 1]
```diff
@@ -942,13 +949,31 @@ public void setIsChanged(boolean changed)
...
+	 * @deprecated - use {@link #getRequestCount(int, int, String, StringBuilder, String)} instead
 	 */
 	public static int[] getRequestCount(int AD_Table_ID, int Record_ID, StringBuilder whereClause, String trxName) {
+		return getRequestCount(AD_Table_ID, Record_ID, null, whereClause, trxName);
+	}
+
+	/**
+	 * Get number of active and inactive request
+	 * @param AD_Table_ID
+	 * @param Record_ID Record ID
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/MRequest.java:961]: the javadoc should be updated to reflect when this is use

File: org.adempiere.base/src/org/compiere/model/MRequest.java [hunk 2]
```diff
@@ -1006,11 +1031,25 @@ else if (AD_Table_ID == MAsset.Table_ID)
...
+
+	/**
+	 * Create new request
+	 * @param tab Grid Tab for request
+	 * @param AD_Table_ID
+	 * @param Record_ID Record ID
+	 * @param Record_UU Record UUID
+	 * @param C_BPartner_ID
+	 */
+	public static void newRequest(GridTab tab, int AD_Table_ID, int Record_ID, String Record_UU, int C_BPartner_ID) {
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/MRequest.java:1048]: This is minor but all this recent changes actually make many changed API harder to use for developer. It kind of make the dev have to think now - must I always fill up both Record_ID and Record_UU parameter ? What happens when I fill up only one of them ? And are there situations where I must fill up both ?

**Comment by CarlosRuiz-globalqss** [comment]
Hi @hengsin - I changed the javadoc to clarify the usage of Record_ID parameter

**Review by hengsin — APPROVED** [review]
tested

**Pull request merged**